### ISSUE TITLE
Error: keyNotFound(CodingKeys(stringValue: "choices", intValue: nil)

### ISSUE THREAD

**Issue by finestructure** [gonzalonunez/docc-gpt#9]
I just went to try this with the [Swift Package Index Server repo](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server) and it throws the following error after processing what seems to be the last file in the list:

```
✓ Finished documenting BuildIndex+Model.swift
✓ Finished documenting BuildShow+View.swift
✓ Finished documenting BuildShow+Model.swift
✓ Finished documenting PackageResult+ModelSupport.swift
✓ Finished documenting PackageShow+StructuredDataSchema.swift
✓ Finished documenting PackageShow+Types.swift
✓ Finished documenting routes.swift
Error: keyNotFound(CodingKeys(stringValue: "choices", intValue: nil), Swift.DecodingError.Context(codingPath: [], debugDescription: "No value associated with key CodingKeys(stringValue: \"choices\", intValue: nil) (\"choices\").", underlyingError: nil))
```

I believe `routes.swift` is the last one, judging by the logs earlier in the list

```
᠅ Documenting PackageShow+StructuredDataSchema.swift...
᠅ Documenting PackageShow+Types.swift...
᠅ Documenting routes.swift...
✓ Finished documenting UpdateRecentPackages1.swift
✓ Finished documenting UpdateVersionAddUrl.swift
✓ Finished documenting UpdateVersionAddPublisedAtReleaseNotes.swift
✓ Finished documenting CreateTarget.swift
```

I wonder if this is due to API metering or something? I'm using the free plan at the moment, while playing around.

Might try this in a smaller project but if you've got a pointer, that'd be great :)

**Comment by finestructure** [comment]
Ah, that's great, thanks for taking a look so quickly!

**Comment by gonzalonunez** [comment]
@finestructure No problem! Still noodling on this, I think most files in the repo are too long but I'll put out a new release with tweaks once I confirm.

I'll then probably try to see what happens with GPT-4.

**Comment by gonzalonunez** [comment]
@finestructure just merged #10, which should make things a little clearer. That said, it doesn't solve your problem––most of the files in the repo are too long as-is. I'm going to try a few things:

1. A new prompt together with GPT-4 and `/v1/chat/completions`. Right now, the prompt requires the entire file to be fed in to it as well as other examples, which means that unless the file fits into the context window together with the prompt _twice_ (as well as with a little bit of extra room for new comments), it won't work or you're at least unlikely to get good results.

2. If that doesn't work, I'll probably try fine-tuning `text-davinci-003` so that we can use the entire context window for the file.

All of that said, I ran the tool on `/Sources/App/Core/Extensions`. Attached you'll find the `.patch`, which has a handful of incorrect changes! Looks like it struggles with header comments too, I should be able to fix this with one of the two strategies above.

The Swift Package Index Server repo is a great test bed for me, I'll continue to experiment on it while I make some tweaks. Sorry things aren't working right out of the gate, but I do think fixes are within reach let's see how this goes.

Thank you for flagging!

[extensions.patch](https://github.com/gonzalonunez/docc-gpt/files/11094169/extensions.patch)

**Comment by finestructure** [comment]
That's great, thanks a lot!
